### ISSUE TITLE
Centraldashboard 403 error 

### ISSUE THREAD

**Issue by y4roslav** [kubeflow/kubeflow#4540]
/kind bug

We are using 0.7 kubeflow with LDAP/Dax integration. In one day our dashboard start rejecting our request with error 403. 

**What did you expect to happen:**
Central Dashboard with Dax page

**Anything else you would like to add:**

Only errors I can see is from central dashboard pod 
```
> node dist/server.js
Initializing Kubernetes configuration
Unable to fetch Nodes { kind: 'Status',
  apiVersion: 'v1',
  metadata: {},
  status: 'Failure',
  message:
   'nodes is forbidden: User "system:serviceaccount:kubeflow:centraldashboard" cannot list resource "nodes" in API g
roup "" at the cluster scope',
  reason: 'Forbidden',
  details: { kind: 'nodes' },
  code: 403 }
Unable to fetch Application information: { kind: 'Status',
  apiVersion: 'v1',
  metadata: {},
  status: 'Failure',
  message:
   'applications.app.k8s.io is forbidden: User "system:serviceaccount:kubeflow:centraldashboard" cannot list resourc
e "applications" in API group "app.k8s.io" in the namespace "kubeflow"',
  reason: 'Forbidden',
  details: { group: 'app.k8s.io', kind: 'applications' },
  code: 403 }
Unable to fetch Nodes { kind: 'Status',
  apiVersion: 'v1',
  metadata: {},
  status: 'Failure',
  message:
   'nodes is forbidden: User "system:serviceaccount:kubeflow:centraldashboard" cannot list resource "nodes" in API g
roup "" at the cluster scope',
  reason: 'Forbidden',
  details: { kind: 'nodes' },
  code: 403 }
```

**Environment:**

- Kubeflow version: 0.7
- kfctl version: 0.7
- Kubernetes platform: CaaS (SUSE, CRI-O)
- Kubernetes version: v1.15.2
- OS: SLES 15

**Comment by aleb** [comment]
My dashboard also started all of a sudden to return 403.
Standard setup following the instructions in https://www.kubeflow.org/docs/started/k8s/kfctl-existing-arrikto/#before-you-start in a Google Kubernetes cluster:
```
$ gcloud container clusters create NAME \
   --zone ZONE \
   --num-nodes=1 \
   --disk-size=50GB \
   --machine-type=n1-standard-4 #Has 15 GB memory
```

I access for example `https://mydomain.com/test1` and after ~20 seconds I get this 403:
```
$ kubectl logs   -n istio-system  pod/istio-ingressgateway-57f7cfdfdd-fxjdm
[...]
[2019-12-11T10:49:58.959Z] "GET /test1 HTTP/1.1" 403 UAEX "-" 0 0 10001 - "85.195.222.99,10.44.0.17" "... browser details ..." "dfbbaff5-ed4a-9183-a902-4b26af8464c9" "mydomain.com" "-" - - 127.0.0.1:80 127.0.0.1:42440 -
```

No cookies seem to be associated with the domain in the browser.

I see this but I don't think it's related, as nothing new appears when the 403 is returned:
```
$ kubectl logs   -n kubeflow pod/centraldashboard-64c4cb4c6-xl8rq
[...]
Unable to fetch Events for myuser: { kind: 'Status',
  apiVersion: 'v1',
  metadata: {},
  status: 'Failure',
  message:
   'events is forbidden: User "system:serviceaccount:kubeflow:centraldashboard" cannot list resource "events" in API group "" in the namespace "myuser"',
  reason: 'Forbidden',
  details: { kind: 'events' },
  code: 403 }
```

Any idea how to get more info about what's causing this?

**Comment by yanniszark** [comment]
Hi all.
I believe you are hitting this issue: https://github.com/arrikto/oidc-authservice/issues/2
This has been fixed in a later version, which is now being merged into v0.7.
https://github.com/kubeflow/manifests/pull/639

The temporary solution is to delete the authservice pod, which will unblock it.
The permanent solution is to update the authservice image.

**Comment by aleb** [comment]
I deleted the pod, but I still get 403:

```
$ kubectl get pods --all-namespaces | grep authservice                                             
istio-system   authservice-55d474b894-plv25                                  1/1     Running     0          3m10s

$ kubectl delete -n istio-system pod/authservice-55d474b894-plv25   
pod "authservice-55d474b894-plv25" deleted
```

I also tried to delete the gateway, but still the same:

```
$ kubectl delete  -n istio-system  pod/istio-ingressgateway-57f7cfdfdd-fxjdm 
pod "istio-ingressgateway-57f7cfdfdd-fxjdm" deleted
```

Is there an option to set a more verbose log level on this gateway?

**Comment by aleb** [comment]
If I recreate the cluster will that include the latest 0.7 including the kubeflow/manifests#639 fix?

**Comment by y4roslav** [comment]
@aleb I applied fix recommended by @yanniszark https://github.com/arrikto/oidc-authservice/issues/2 and looks like it woks. To check that your build would work check you kustromize files. Both should contain this lines:

istio/oidc-authservice/base/deployment.yaml

```
readinessProbe:
             httpGet:
               path: /
               port: 8081
```

and major one `istio/oidc-authservice/base/kustomization.yaml` should have tag `newTag: 28c59ef` instead  `6ac9400` 

It works for my installation. 

PS If it keep working next few days and you would be able to fix it I will close this issue.

**Comment by aleb** [comment]
Until https://www.kubeflow.org/docs/started/k8s/kfctl-existing-arrikto is updated to point to an updated yaml, I'll use the workaround. It was not enough to delete the pod, I had to do it like this:

```
$ kubectl delete -n istio-system deployment.apps/authservice
$ ~/Downloads/kfctl-darwin apply -V -f kfctl_existing_arrikto.yaml 
$ kubectl describe -n istio-system pod/authservice-6bcdbbd76d-crl5q | grep Readiness
    Readiness:      http-get http://:8081/ delay=0s timeout=1s period=10s #success=1 #failure=3
```
